pyproject.toml

```diff
@@ -189,6 +189,23 @@ ignore = [
     # Fix these codes later
     "G004",
     "PERF203",
+    # https://github.com/astral-sh/ruff/issues/7871
+    "UP038",
```

Comment:
This should be disabled, but it should also be tabled by setting the target python version low enough.

Reply:
Could you provide more precise guidance? Right now, the minimal supported version is defined in  `pyproject.toml`:
```toml
requires-python = ">=3.8"
````
What would you like to change? Also, why should the minimal supported version of Python depend on a ruff rule?

Reply:
@Skylion007 , are you OK with having "UP038" in this list for now?

Reply:
Leave it disabled, it doesn't affect us anyway. 

If it's really triggering despite the Python version, this might be a ruff a bug that we should file upstream though. It's suppose to have version checks to prevent it suggesting upgrades which break the min version.